### PR TITLE
Make mariadb's service supports NodePort mode

### PR DESCRIPTION
**Pull request by zachzhaozy**
In my case, the MariaDB which is deployed on kubernetes is going to be used by apps outside kubernetes, so I did such things to achieve the goal:

- Move `service` to each of master and slave's config content
- Add new service type: `NodePort`
- Update `NOTES.txt`
- Update `values.yaml` and `values-production.yaml`

### CODE REVIEW

**Comment by zacharyzhao** [comment]
I move `MariaDB` service configuration to each of master and slave's config block, so their `NodePort` can be different, then apps running outside k8s can use the different port for writing or reading. @sameersbn @juan131

**Review by juan131 — CHANGES_REQUESTED** [review]
@zacharyzhao moving `service` to each of master and slave's configure content breaks compatibility with previous versions and, therefore, you can't upgrade from previous versions... I don't see any advantage on this and.. since it means a major version of the chart. I'd try to avoid it.

On the other hand, I agree with including the possibility of configuring the `nodePort`.

**Comment by zacharyzhao** [comment]
@juan131  I updated this PR, I put service block back and add configures for `NodePort` in it.

File: stable/mariadb/templates/master-svc.yaml [hunk 1]
```diff
@@ -18,6 +18,11 @@ spec:
   - name: mysql
     port: {{ .Values.service.port }}
     targetPort: mysql
+{{- if eq .Values.service.type "NodePort" }}
```
Comment (juan131) [inline stable/mariadb/templates/master-svc.yaml:21]: You can use a single line:

```
{{- if (and (eq .Values.serviceType "NodePort") (not (empty .Values.service.nodePort.master))) }}
```

File: stable/mariadb/templates/slave-svc.yaml [hunk 2]
```diff
@@ -19,6 +19,11 @@ spec:
   - name: mysql
     port: {{ .Values.service.port }}
     targetPort: mysql
+{{- if eq .Values.service.type "NodePort" }}
```
Comment (juan131) [inline stable/mariadb/templates/slave-svc.yaml]: You can use a single line:
```
{{- if (and (eq .Values.serviceType "NodePort") (not (empty .Values.service.nodePort.slave))) }}
```

File: stable/mariadb/values-production.yaml [hunk 3]
```diff
@@ -21,6 +21,9 @@ service:
   ## Kubernetes service type
   type: ClusterIP
   port: 3306
+  nodePort:
```
Comment (juan131) [inline stable/mariadb/values-production.yaml]: Great! Please add some comments:

```
  ## Specify the nodePort value for the LoadBalancer and NodePort service types.
  ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
  ##
  nodePort:
  ...
```

**Review by juan131 — COMMENTED** [review]
Much better! Please take a look to my suggestions

**Comment by zacharyzhao** [comment]
Hi, @juan131 , thanks for your advises, and I've already updated the branch, please check if it is ok now.

Thank you very much!~

**Pull request merged**